component/src/main/java/org/wso2/extension/siddhi/io/ibmmq/source/IBMMQSource.java

```diff
@@ -0,0 +1,208 @@
+                !"null".equalsIgnoreCase(optionHolder.validateAndGetStaticValue(IBMMQConstants.PASSWORD, "null"))) {
+            isSecured = true;
+            userName = optionHolder.validateAndGetOption(IBMMQConstants.USER_NAME).getValue();
+            password = optionHolder.validateAndGetOption(IBMMQConstants.PASSWORD).getValue();
+        }
+        try {
+            connectionFactory.setChannel(optionHolder.validateAndGetOption(IBMMQConstants.CHANNEL).getValue());
+            connectionFactory.setHostName(optionHolder.validateAndGetOption(IBMMQConstants.HOST).getValue());
+            connectionFactory.setPort(Integer.parseInt(optionHolder.
+                    validateAndGetOption(IBMMQConstants.PORT).getValue()));
+            connectionFactory.setQueueManager(optionHolder.validateAndGetOption(IBMMQConstants.QUEUE_MANAGER_NAME)
+                    .getValue());
+            connectionFactory.setTransportType(1);
+            scheduledExecutorService = siddhiAppContext.getScheduledExecutorService();
+        } catch (JMSException e) {
+            throw new IBMMQInputAdaptorRuntimeException("Error while initializing IBM MQ source: " + e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    public void connect(ConnectionCallback connectionCallback) throws ConnectionUnavailableException {
+        //ConnectionCallback is not used as re-connection is handled by carbon transport.
+        try {
+            if (isSecured) {
+                connection = (MQConnection) connectionFactory.createConnection(userName, password);
+            } else {
+                connection = (MQConnection) connectionFactory.createConnection();
+            }
+            session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            queue = session.createQueue(optionHolder.validateAndGetOption(IBMMQConstants.DESTINATION_NAME)
+                    .getValue());
+            consumer = session.createConsumer(queue);
+            ibmMessageConsumer = new IBMMessageConsumer(sourceEventListener, connection, consumer);
+            ibmMessageConsumer.consume(ibmMessageConsumer, scheduledExecutorService);
+
+        } catch (JMSException e) {
+            throw new ConnectionUnavailableException("Exception in starting the IBM MQ receiver for stream: "
+                    + sourceEventListener.getStreamDefinition().getId(), e);
+        }
+    }
+
+    @Override
+    public Class[] getOutputEventClasses() {
+        return new Class[]{String.class, Map.class};
+    }
+
+    @Override
+    public void disconnect() {
+        try {
+            if (Objects.nonNull(connection)) {
+                connection.close();
+                ibmMessageConsumer.kill();
+            }
+            if (Objects.nonNull(consumer)) {
+                consumer.close();
+            }
+        } catch (JMSException e) {
+            LOG.error("Error disconnecting the IBM MQ connection ", e);
```

Comment:
Shall we add some useful info to identify the specific ibm mq source here? (queue/topic name  etc)